rest-new/src/main/java/org/jboss/pnc/rest/endpoints/GroupBuildEndpointImpl.java

```diff
@@ -0,0 +1,114 @@
+/**
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2014-2019 Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jboss.pnc.rest.endpoints;
+
+import java.lang.invoke.MethodHandles;
+
+import javax.inject.Inject;
+
+import org.jboss.pnc.rest.api.parameters.BuildsFilterParameters;
+import org.jboss.pnc.rest.api.parameters.PageParameters;
+
+import org.jboss.pnc.dto.Build;
+import org.jboss.pnc.dto.GroupBuild;
+import org.jboss.pnc.dto.GroupBuildRef;
+import org.jboss.pnc.dto.requests.GroupBuildPushRequest;
+import org.jboss.pnc.dto.response.Graph;
+import org.jboss.pnc.dto.response.Page;
+import org.jboss.pnc.facade.providers.api.BrewPusher;
+import org.jboss.pnc.facade.providers.api.BuildConfigurationProvider;
+import org.jboss.pnc.facade.providers.api.BuildProvider;
+import org.jboss.pnc.facade.providers.api.GroupBuildProvider;
+import org.jboss.pnc.rest.api.endpoints.GroupBuildEndpoint;
+import static org.jboss.pnc.rest.endpoints.BuildEndpointImpl.toBuildPageInfo;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.ws.rs.NotFoundException;
+
+/**
+ *
+ * @author Honza Brázdil &lt;jbrazdil@redhat.com&gt;
+ */
+@ApplicationScoped
```

Comment:
`@Stateless` maybe?

Reply:
Why? It doesn't have to be in transactions.

Reply:
Wouldn't @ApplicationScoped create a singleton basically?

Reply:
Yes it would. That's why I'm using it.

Reply:
Does that affect the throughput when requests go in? If I remember correctly, its 1 thread per request? But yeah I dunno how this behaves with CDI and no concurrency limit.